src/app/clusters/mode-select-server/static-supported-modes-manager.h

```diff
@@ -36,97 +40,33 @@ namespace ModeSelect {
-    using storage_value_type   = const ModeOptionStructType *;
-
-    static const ModeOptionStructType blackOption;
-    static const ModeOptionStructType cappuccinoOption;
-    static const ModeOptionStructType espressoOption;
-
-    static const ModeOptionStructType * coffeeOptions[];
-    static const Span<storage_value_type> coffeeOptionsSpan;
-    static const std::map<EndpointId, Span<storage_value_type>> optionsByEndpoints;
-
-public:
-    static const StaticSupportedModesManager instance;
+    using storage_value_type   = const ModeOptionStructType;
 
-    struct Iterator : public chip::app::Clusters::ModeSelect::SupportedModesManager::ModeOptionStructIterator
+    struct EndpointSpanPair
     {
-        using iterator_category  = std::forward_iterator_tag;
-        using difference_type    = std::ptrdiff_t;
-        using pointer            = storage_value_type *;
-        using base_iterator_type = chip::app::Clusters::ModeSelect::SupportedModesManager::ModeOptionStructIterator;
-
-        Iterator(const pointer aPtr) : mPtr(aPtr) {}
-        ~Iterator() = default;
-
-        const ModeOptionStructType & operator*() const override { return **mPtr; }
-        const ModeOptionStructType * operator->() override { return *mPtr; }
-        const ModeOptionStructType * operator->() const override { return *mPtr; }
-
-        // Prefix increment
-        base_iterator_type & operator++() override
-        {
-            ++mPtr;
-            return *this;
-        }
-
-        bool operator==(const base_iterator_type & other) const override
-        {
-            // Warning: we are not doing type check
-            // TODO: use of typeid requires -frtti
-            // if (typeid(other) != typeid(*this))
-            // {
-            //     return false;
-            // }
-            return this->operator->() == other.operator->();
-        }
-        bool operator!=(const base_iterator_type & other) const override { return !((*this) == other); }
-
-    private:
-        pointer mPtr;
-    };
+        const EndpointId mEndpointId;
+        const Span<storage_value_type> mSpan;
 
-    struct IteratorFactory : public chip::app::Clusters::ModeSelect::SupportedModesManager::ModeOptionStructIteratorFactory
-    {
-        using pointer       = Iterator *;
-        using const_pointer = const pointer;
-
-        IteratorFactory(const Span<storage_value_type> & aSupportedOptions) :
-            _begin(Iterator(aSupportedOptions.data())), _end(Iterator(aSupportedOptions.data() + aSupportedOptions.size()))
+        EndpointSpanPair(const EndpointId aEndpointId, const Span<storage_value_type> && aSpan) :
+            mEndpointId(aEndpointId), mSpan(aSpan)
         {}
-        ~IteratorFactory() = default;
+    };
 
-        const Iterator * begin() const override { return &_begin; }
-        const Iterator * end() const override { return &_end; }
+    static storage_value_type coffeeOptions[];
```

Comment:
This really seems like a concrete detail that should be part of a unit test only